nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RecordStats.java

```diff
@@ -0,0 +1,176 @@
+        .build();
+
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder()
+            .name(propertyDescriptorName)
+            .displayName(propertyDescriptorName)
+            .dynamic(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+    }
+
+    private RecordPathCache cache;
+
+    @OnScheduled
+    public void onEnabled(ProcessContext context) {
+        cache = new RecordPathCache(25);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return new HashSet<Relationship>() {{
+            add(REL_SUCCESS);
+            add(REL_FAILURE);
+        }};
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile input = session.get();
+        if (input == null) {
+            return;
+        }
+
+        try {
+            Map<String, RecordPath> paths = getRecordPaths(context);
+            Map<String, String> stats = getStats(input, paths, context, session);
+
+            input = session.putAllAttributes(input, stats);
+
+            session.transfer(input, REL_SUCCESS);
+
+        } catch (Exception ex) {
+            getLogger().error("Error processing stats.", ex);
+            session.transfer(input, REL_FAILURE);
+        }
+
+    }
+
+    protected Map<String, RecordPath> getRecordPaths(ProcessContext context) {
+        return context.getProperties().keySet()
+            .stream().filter(p -> p.isDynamic() && !p.getName().contains(RECORD_READER.getName()))
+            .collect(Collectors.toMap(
+                e -> e.getName(),
+                e ->  {
+                    String val = context.getProperty(e).getValue();
+                    return cache.getCompiled(val);
+                })
+            );
+    }
+
+    protected Map<String, String> getStats(FlowFile flowFile, Map<String, RecordPath> paths, ProcessContext context, ProcessSession session) {
+        try (InputStream is = session.read(flowFile)) {
+            RecordReaderFactory factory = context.getProperty(RECORD_READER).asControllerService(RecordReaderFactory.class);
+            RecordReader reader = factory.createRecordReader(flowFile, is, getLogger());
+
+            Map<String, Integer> retVal = new HashMap<>();
+            Record record;
+
+            int recordCount = 0;
+            while ((record = reader.nextRecord()) != null) {
+                for (Map.Entry<String, RecordPath> entry : paths.entrySet()) {
+                    RecordPathResult result = entry.getValue().evaluate(record);
+                    Optional<FieldValue> value = result.getSelectedFields().findFirst();
+                    if (value.isPresent() && value.get().getValue() != null) {
+                        String approxValue = value.get().getValue().toString();
```

Comment:
Idea for improvement. If the RecordPath result is number, providing the counts based on value will be useful? Rather, I'd like to see in addition to counts (current `baseStat`), also min, max, sum and optionally sum of squared x.

Reply:
Let's do that as a separate ticket if you don't mind.

Reply:
Yes, we can add this later.

I hoped it will be included when this processor is released. Changing it after would require adding new configuration property. Well, that makes sense in some cases, too. E.g. if a number field represents some category information, or act as an enum, then user would expected the number of occurrence per the number value.